Merge changelogs

RobinMalfait opened the pull request:
This PR makes sure all v3 and earlier versions are visible in the changelog as
well as the v4 changelog entries.

RobinMalfait committed:
merge changelogs

philipp-spiess commented on CHANGELOG.md:
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Nothing yet
 
 ## [4.0.0] - 2025-01-21
+
```
Feels weird that this is empty, hmm. Should we include something here?

thecrypticace replied:
Should we maybe repeat (some of?) the info from the GitHub release?

RobinMalfait replied:
Added some of the contents of the GitHub release.

I also added compare links for the individual alpha/beta releases (but before v4 alpha 21, we didn't publish GitHub releases so those aren't included)

RobinMalfait committed:
add missing links

There are more missing links for v4.0.0-alpha.21 and lower, but we don't
have GitHub releases for them. We could link to the actual commits, but
that might take a bit more work, and not sure if that's really useful.

RobinMalfait committed:
fill in `4.0.0` release

This is using content from the GitHub release notes.
